src/richie/apps/courses/templates/courses/cms/category_detail.html

```diff
@@ -92,6 +92,17 @@ <h2 class="course-glimpse-list__title">{% trans "Related courses" %}</h2>
           {% include "courses/cms/fragment_course_glimpse.html" %}
         {% endif %}
       {% endfor %}
+      {% if courses.count == PAGES_SETTINGS.MAX_RELATED_COURSES_ON_DETAILS_PAGES %}
+        {% if category.get_meta_category %}
```

Comment:
why not show the search page filtered on this category? It seems to be what you naturally want to see and not all sibling categories.

Reply:
I'm not sure what you're suggesting here. It seems to me the link is precisely pointing to `the search page filtered on this category`

Reply:
ok! I just realized why you need to get the id of the meta to do that :-/